api/src/controllers/resumeReview/patchAllResumeReview.ts

```diff
@@ -0,0 +1,64 @@
+import { Request, Response } from 'express';
+import type * as s from 'zapatos/schema';
+
+import InternalServerErrorException from '../../exceptions/InternalServerErrorException';
+import * as resumeReviewRepository from '../../repositories/resumeReviewRepository';
+import controller from '../controllerUtil';
+import Validator, { beAResumeReviewState, beAValidResumeReview, beAValidUser, beAValidUuid } from '../validation';
+
+type Params = {
+    resumeReview: string;
+};
+
+class ParamsValidator extends Validator<Params> {
+    constructor() {
+        super('route parameters');
+
+        this.ruleFor('resumeReview').mustAsync(beAValidUuid).mustAsync(beAValidResumeReview);
+    }
+}
+
+type ReqBody = {
+    reviewee?: string;
+    reviewer?: string;
+    state?: s.resume_review_state;
+};
+
+class ReqBodyValidator extends Validator<ReqBody> {
+    constructor() {
+        super('message body');
+
+        this.ruleFor('reviewee')
+            .mustAsync(beAValidUser)
+            .when((reqBody) => reqBody.reviewee !== undefined);
```

Comment:
Is this preferred over

```ts
        this.ruleFor('reviewee').notNull().notEmpty();
```
?

Reply:
@armiantos Does something entirely different actually. Forces the validation to only run when the field is defined. If we didn't include this the validation would fail if you don't include a `reviewee` field which is something we don't want because it is possible that they don't want to update the reviewee.

[Doc](https://fluentvalidation-ts.alexpotter.dev/docs/api/configuration/when)

Reply:
Gotcha! Thanks for the clarification